guard/src/rules/functions/date_time_tests.rs

```diff
@@ -0,0 +1,92 @@
+use std::{convert::TryFrom, rc::Rc};
+
+use crate::rules::{
+    eval_context::eval_context_tests::BasicQueryTesting,
+    exprs::AccessQuery,
+    functions::date_time::{now, parse_epoch},
+    path_value::PathAwareValue,
+    EvalContext, QueryResult,
+};
+use chrono::Utc;
+use pretty_assertions::assert_eq;
+use rstest::rstest;
+
+const VALUE_STR: &str = r#"
+{
+    "Resources": {
+        "LambdaFunction": {
+            "Type": "AWS::Lambda::Function",
+            "Properties": {
+                "UpdatedAt": "2024-08-21T00:00:00Z",
+                "CreatedAt": "2024-08-13T00:00:00Z",
+                "BadValue": "not-a-date"
+            }
+        }
+    }
+}
+    "#;
+
+#[rstest]
+#[case(r#"UpdatedAt"#, 1724198400, false)]
+#[case(r#"CreatedAt"#, 1723507200, false)]
+#[case(r#"BadValue"#, 1723507200, true)]
+fn test_parse_epoch(
+    #[case] query_str: &str,
+    #[case] _expected_epoch: i64,
+    #[case] should_error: bool,
+) -> crate::rules::Result<()> {
+    let value = PathAwareValue::try_from(serde_yaml::from_str::<serde_yaml::Value>(VALUE_STR)?)?;
```

Comment:
nit: should use `serde_json` since we're passing a json string in this example. We can also use serde_json::to_value to increase readability here

Reply:
ack will do!